frontend/src/store/playbackManager.ts

```diff
@@ -1092,25 +1157,33 @@ class PlaybackManagerStore {
      */
     watch(
       () => this.currentItemIndex,
-      async () => {
-        const playbackInfo = await this.getItemPlaybackInfo();
+      async (newIndex, oldIndex) => {
+        await this.fetchCurrentMediaSource();
 
-        if (playbackInfo) {
-          const mediaSource = playbackInfo.MediaSources?.[0];
+        const router = useRouter();
 
-          if (!mediaSource) {
-            const { t } = usei18n();
+        if (newIndex && !state.currentSourceUrl) {
+          const { t } = usei18n();
 
-            useSnackbar(t('errors.cantPlayItem'), 'error');
-          } else {
-            state.playSessionId = playbackInfo?.PlaySessionId || '';
-            state.currentMediaSource = mediaSource;
-            state.currentSourceUrl = this.getItemPlaybackUrl();
-          }
+          useSnackbar(t('errors.cantPlayItem'), 'error');
+          this.stop();
+        } else if (
+          this.currentlyPlayingMediaType === 'Video' &&
+          oldIndex === undefined &&
+          newIndex !== undefined
+        ) {
+          router.push('/playback/video');
+        } else if (
+          router.currentRoute.value.fullPath === '/playback/video' &&
+          newIndex === undefined
+        ) {
+          router.replace(
+            typeof router.options.history.state.back === 'string'
+              ? router.options.history.state.back
+              : '/'
+          );
```

Comment:
This should be handled by a watcher in the fullscreen page, as commented above

Reply:
Moved to playerElement